Fix bug in JoinGraphBuilder

mrzzzrm opened the pull request:
Fix a bug in the `JoinGraphBuilder` where subgraphs not connected by any predicates would leave the `JoinGraph` unconnected. This PR makes the `JoinGraphBuilder` insert unpredicated edges between the components to create a connected `JoinGraph`

mrks commented on src/lib/optimizer/join_ordering/join_graph_builder.cpp:
```diff
@@ -169,4 +177,109 @@ JoinGraphBuilder::PredicateParseResult JoinGraphBuilder::_parse_union(
+    bfs_stack.push(vertex_idx);
+
+    while (!bfs_stack.empty()) {
+      const auto vertex_idx2 = bfs_stack.top();
```
Can we find a better name for the two indexes?

mrks commented on src/lib/optimizer/join_ordering/join_graph_builder.cpp:
```diff
@@ -169,4 +177,109 @@ JoinGraphBuilder::PredicateParseResult JoinGraphBuilder::_parse_union(
+
+    while (!bfs_stack.empty()) {
+      const auto vertex_idx2 = bfs_stack.top();
+      bfs_stack.pop();
```
Breadth-First? I think I could live with the abbreviation, but a comment would be good.

mrks commented on src/lib/optimizer/join_ordering/join_graph_builder.cpp:
```diff
@@ -169,4 +177,109 @@ JoinGraphBuilder::PredicateParseResult JoinGraphBuilder::_parse_union(
+  for (auto vertex_idx = size_t{0}; vertex_idx < vertices.size(); ++vertex_idx)
+    remaining_vertex_indices.insert(vertex_idx);
+
+  std::vector<size_t> one_vertex_per_component;
```
What does this store?

mrzzzrm commented:
Addressed your comments and then some. The long time it takes to get an review has the pro side that you're forced to look at older code and then understand it again. Added quite some comments and improved naming.

mrks commented:
Haha, you don't want to provide me with an excuse for leaving these for a longer time ;)
